### PR TITLE
Try to find interface of class when apply mapper

### PR DESCRIPTION
**Pull request by artem-zinnatullin**
Little bit refactored version of PR #580  from @omihaz 

Splitted tests into 3 separe ones and removed unneeded merge commit (rebased)

@nikitin-da PTAL (@omihaz you can take a look too :smile:)

### CODE REVIEW

**Comment by artem-zinnatullin** [comment]
// @omihaz don't worry, you're still author of the commit

File: storio-sqlite/src/test/java/com/pushtorefresh/storio/sqlite/impl/DefaultStorIOSQLiteTest.java [hunk 1]
```diff
@@ -242,6 +243,65 @@ public void typeMappingShouldFindIndirectTypeMappingInCaseOfComplexInheritance()
...
+        }
+
+        assertThat(storIOSQLite.internal().typeMapping(ConcreteEntity.class)).isSameAs(typeMapping);
+
+        // Just to make sure that we don't return this type mapping for all classes.
+        assertThat(storIOSQLite.internal().typeMapping(Random.class)).isNull();
+    }
+
+    @Test
+    public void typeMappingShouldFindIndirectTypeMappingForClassThatHasParentThatImplementsKnownInterface() {
```
Comment (nikitin-da) [inline storio-sqlite/src/test/java/com/pushtorefresh/storio/sqlite/impl/DefaultStorIOSQLiteTest.java]: Nice name)

**Comment by omihaz** [comment]
Okay, thanks :)

**Comment by nikitin-da** [comment]
LGTM 
Сan you please rebase it again?)

**Comment by artem-zinnatullin** [comment]
Oh shi, ok, ok

**Comment by artem-zinnatullin** [comment]
Rebased, @nikitin-da please merge after CI, I'll be away from computer for several hours :)

**Comment by nikitin-da** [comment]
No problem. Thank you @omihaz and @artem-zinnatullin

**Pull request merged**